### PR TITLE
Added TLS configurations and support

### PR DESCRIPTION
**Pull request by avanishp**

### CODE REVIEW

**Review by koushikr — COMMENTED** [review]
Do you want to leave a validation here if tlsConfig is set to true then the trustStore paths shouldn't be null or empty?

File: src/main/java/io/github/jelastic/core/config/AuthConfiguration.java [hunk 1]
```diff
@@ -33,4 +33,10 @@
 
     @NotNull @NotEmpty
     private String password;
+
+    private boolean tlsEnabled;
+
+    private String trustStorePath;
```
Comment (koushikr) [inline src/main/java/io/github/jelastic/core/config/AuthConfiguration.java:40]: Do you want to leave a validation here if tlsConfig is set to true then the trustStore paths shouldn't be null or empty?

File: src/main/java/io/github/jelastic/core/elastic/ElasticClient.java [hunk 2]
```diff
@@ -76,6 +87,20 @@ public ElasticClient(JElasticConfiguration configuration) throws IOException {
...
+            if(jElasticConfiguration.getAuthConfiguration().isTlsEnabled()){
+                Path trustStorePath = Paths.get(jElasticConfiguration.getAuthConfiguration().getTrustStorePath());
+                KeyStore truststore = KeyStore.getInstance("pkcs12");
+                try (InputStream is = Files.newInputStream(trustStorePath)) {
+                    truststore.load(is, jElasticConfiguration.getAuthConfiguration().getKeyStorePass().toCharArray());
+                }
+                SSLContextBuilder sslBuilder = SSLContexts.custom()
+                        .loadTrustMaterial(truststore, null);
+                final SSLContext sslContext = sslBuilder.build();
+                restClientBuilder.setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder.setSSLContext(sslContext));
```
Comment (koushikr) [inline src/main/java/io/github/jelastic/core/elastic/ElasticClient.java]: You can combine both the lines setSSLContext and restClientBuilder.setCredentials into one lamdda, and declare the variables atop with a null check. Possibly extract out these two into different methods getSSLContext and getCredentials; and if they are not null, then apply to the clientBuilder.

Comment (avanishp) [reply src/main/java/io/github/jelastic/core/elastic/ElasticClient.java]: this has been taken care of !

**Pull request merged**